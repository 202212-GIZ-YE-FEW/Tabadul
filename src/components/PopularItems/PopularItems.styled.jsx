import styled from "@emotion/styled";
import Image from "next/image";

export const Header = styled.h1`
    text-align: center;
    font-family: "Poppins";
    font-style: normal;
    font-weight: 700;
    font-size: 56px;
    line-height: 84px;
    color: #3c4347;
`;

export const Cards = styled.div`
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    gap: 20px;
`;

export const Card = styled.div`
    // position: absolute;
    background: #ffffff;
    box-shadow: 0px 0px 7px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    background: #ffffff;
`;

export const Cardimg = styled(Image)`
    border-radius: 19px 19px 0px 0px;
`;

export const Cardinfo = styled.div`
    display: flex;
    align-items: center;
    margin-left: 10px;
`;

export const Icon = styled(Image)`
    margin-right: 10px &.icon {
        width: 32px;
        height: 32px;
    }
`;
export const Cardcontainer = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-start;
    margin-bottom: 20px;
`;

export const LocationTitle = styled.p`
    font-family: "Poppins";
    font-weight: 400;
    font-size: 20px;
    line-height: 30px;
    text-align: center;
    margin-left: 10px;
`;

export const CardHeader = styled.p`
    font-family: "Poppins";
    font-weight: 700;
    font-size: 20px;
    line-height: 30px;
    color: #3c4347;
    margin-left: 10px;
    margin-top: 0;
`;
export const CardParagraph = styled.p`
    font-family: "Poppins";
    font-size: 16px;
    font-weight: 400;
    width: 200px;
    color: #3c4347;
    margin-left: 10px;
    margin-top: 0;
`;

export const Line = styled.hr`
    // position: absolute;
    // width: 1701px;
    // height: 0px
    // // left: 78px;
    // // top: 3251px;
    // border: 1px solid #33956D;
    // color:background: rgba(51, 149, 109, 1);

    position: relative;
    left: 50%;
    // bottom: 3%;
    transform: translate(-50%, -50%);
    width: 95%;
    height: 0px;
    background-color: #33956d;
    margin-top: 40px;
`;
