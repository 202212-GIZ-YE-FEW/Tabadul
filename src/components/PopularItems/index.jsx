import React from "react";

import {
    Card,
    Cardcontainer,
    CardHeader,
    Cardimg,
    Cardinfo,
    CardParagraph,
    Icon,
    Lo_Ca_Div,
    LocationTitle,
    ProductTitle,
} from "./PopularItems.styled";
import { DateAndTime } from "../ListedItems/ListedItems.styled";

function PopularItems({ category, location, photo, description, title, date }) {
    return (
        <Card>
            <Cardimg
                src={photo}
                alt='CardImage'
                width={258.4}
                height={185.44}
            />
            <Cardcontainer>
                <ProductTitle>
                    {title?.length > 50
                        ? `${title.substring(0, 50)} ...`
                        : title}
                </ProductTitle>
                <Lo_Ca_Div>
                    <Cardinfo>
                        <Icon
                            src='/images/Vector.svg'
                            alt='Icon'
                            width={17.77}
                            height={20.61}
                        />
                        <LocationTitle>{location}</LocationTitle>
                    </Cardinfo>
                    <Cardinfo>
                        <Icon
                            src='/images/tag-fill.svg'
                            alt='Icon'
                            width={14}
                            height={20}
                        />
                        <LocationTitle>{category}</LocationTitle>
                    </Cardinfo>
                </Lo_Ca_Div>
                <DateAndTime>{date}</DateAndTime>
                <CardParagraph>
                    {description?.length > 50
                        ? `${description.substring(0, 50)} ...`
                        : description}
                </CardParagraph>
            </Cardcontainer>
        </Card>
    );
}

export default PopularItems;
