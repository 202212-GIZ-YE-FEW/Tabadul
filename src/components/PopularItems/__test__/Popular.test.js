import renderer from "react-test-renderer";
import PopularItems from "..";

it("renders correctly", () => {
    const tree = renderer.create(<PopularItems />).toJSON();
    expect(tree).toMatchSnapshot();
});
