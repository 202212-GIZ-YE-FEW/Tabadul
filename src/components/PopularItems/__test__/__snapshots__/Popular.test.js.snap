// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-ihlp1b"
>
  <img
    alt="CardImage"
    className="css-xwv4wc"
    data-nimg="1"
    decoding="async"
    height={185.44}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src=""
    style={
      Object {
        "color": "transparent",
      }
    }
    width={258.4}
  />
  <div
    className="css-gdod6k"
  >
    <p
      className="css-18ekudf"
    />
    <div
      className="css-65ralr"
    >
      <div
        className="css-11fp3er"
      >
        <img
          alt="Icon"
          className="css-1my2hp2"
          data-nimg="1"
          decoding="async"
          height={20.61}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/images/Vector.svg"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={17.77}
        />
        <p
          className="css-c1ouhe"
        />
      </div>
      <div
        className="css-11fp3er"
      >
        <img
          alt="Icon"
          className="css-1my2hp2"
          data-nimg="1"
          decoding="async"
          height={20}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/images/tag-fill.svg"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={14}
        />
        <p
          className="css-c1ouhe"
        />
      </div>
    </div>
    <p
      className="css-1eccmhm"
    />
    <p
      className="css-1c9mivk"
    />
  </div>
</div>
`;
